Extension/src/LanguageServer/ui.ts

```diff
@@ -58,288 +58,266 @@ const commandArguments: string[] = []; // We report the sender of the command
+        this.referencesStatusBarItem = this.createReferencesStatusBarItem();
         this.ShowReferencesIcon = false;
 
-        this.configureIntelliSenseStatusItem = vscode.window.createStatusBarItem(`c.cpp.configureIntelliSenseStatus.statusbar`, vscode.StatusBarAlignment.Right, 0);
-        this.configureIntelliSenseStatusItem.name = this.cppConfigureIntelliSenseText;
-        this.configureIntelliSenseStatusItem.tooltip = this.cppConfigureIntelliSenseText;
-        this.configureIntelliSenseStatusItem.text = `$(warning) ${this.configureIntelliSenseText}`;
-        this.configureIntelliSenseStatusItem.backgroundColor = new vscode.ThemeColor('statusBarItem.warningBackground');
-        this.configureIntelliSenseStatusItem.command = {
-            command: "C_Cpp.SelectIntelliSenseConfiguration",
-            title: this.configureIntelliSenseStatusItem.name,
-            arguments: ['statusBar']
-        };
+        this.configurationStatusBarItem = this.createConfigurationStatusBarItem();
+        this.ShowConfiguration = true;
+
+        this.configureIntelliSenseStatusBarItem = this.createConfigureIntelliSenseStatusBarItem();
         void this.ShowConfigureIntelliSenseButton(false, this.currentClient);
+    }
 
-        this.intelliSenseStatusItem = vscode.languages.createLanguageStatusItem(`cpptools.status.${LanguageStatusPriority.Mid}.intellisense`, util.documentSelector);
-        this.intelliSenseStatusItem.name = localize("cpptools.status.intellisense", "C/C++ IntelliSense Status");
-        this.intelliSenseStatusItem.text = this.idleIntelliSenseText;
+    //#region IntelliSense language status
+    private createIntelliSenseStatusItem(): vscode.LanguageStatusItem {
+        const item: vscode.LanguageStatusItem = vscode.languages.createLanguageStatusItem(`cpptools.status.${LanguageStatusPriority.High}.intellisense`, util.documentSelector);
+        item.name = localize("cpptools.status.intellisense", "C/C++ IntelliSense Status");
+        item.text = this.idleIntelliSenseText;
+        return item;
+    }
 
-        this.browseEngineStatusItem = vscode.languages.createLanguageStatusItem(`cpptools.status.${LanguageStatusPriority.Mid}.tagparser`, util.documentSelector);
-        this.browseEngineStatusItem.name = localize("cpptools.status.tagparser", "C/C++ Tag Parser Status");
-        this.browseEngineStatusItem.detail = localize("cpptools.detail.tagparser", "Initializing...");
-        this.browseEngineStatusItem.text = "$(database)";
-        this.browseEngineStatusItem.command = {
-            command: "C_Cpp.RescanWorkspace",
-            title: this.workspaceRescanText,
+    private flameTimeout?: NodeJS.Timeout;
+    private setIsUpdatingIntelliSense(val: boolean): void {
+        const settings: CppSettings = new CppSettings((vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) ? vscode.workspace.workspaceFolders[0]?.uri : undefined);
+
+        if (settings.intelliSenseEngine === "disabled") {
```

Comment:
Did we determine whether this was dead code?  I was thinking that we don't enable the language status when IntelliSense is disabled.